tests/unit/amo/components/TestAddonQRCodeLink.js

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+
+import AddonQRCode from 'amo/components/AddonQRCode';
+import AddonQRCodeLink, {
+  AddonQRCodeLinkBase,
+} from 'amo/components/AddonQRCodeLink';
+import { CLIENT_APP_ANDROID, CLIENT_APP_FIREFOX } from 'amo/constants';
+import {
+  createFakeEvent,
+  createInternalAddonWithLang,
+  dispatchClientMetadata,
+  fakeAddon,
+  fakeI18n,
+  getFakeConfig,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  let store;
+
+  beforeEach(() => {
+    store = dispatchClientMetadata().store;
+  });
+
+  const goodAddonId = 12345;
```

Comment:
I don't know what a "good" add-on is. Could it be `config.get('addonIdsWithQRCodes')[0]` instead? We wouldn't have to update `addonIdsWithQRCodes` in the fake config. We could use add-on ID `123` when we need an ID without a QR code.

Reply:
Do you really not know what a "good" add-on is? It seems pretty obvious from the code. I am following the convention of being explicit in the test, by setting values into the `config` rather than relying on what's already there. Your suggestion could work for the "good" case (where "good" means an id that is valid for the QR code link), but just using `123` for the "bad" case isn't guaranteed. For example, it's possible that `123` exists in the current `config`. I know that we know it isn't there, but it's *possible*. It seems safer, when writing a test, to use a value that is guaranteed not to be there, and the only way to do that is to know which values are there (by adding them ourselves) and then use a value that we know we didn't add, which is exactly what I am doing.

This is a much more common practice in our test cases (defining the value for the `config` inside the test case, rather than relying on an existing `config` value), and I think it's more of a best practice.

The advantage to your method is that we wouldn't need to do DI for the config, so we could remove that code from the component as well, but again, this is a pretty common practice in our code, so I'm not sure we need to deviate from it here.

Reply:
> Do you really not know what a "good" add-on is?

I know what an add-on with a QR code is.